observation_portal/requestgroups/serializers.py

```diff
@@ -863,16 +862,77 @@ class DitherSerializer(serializers.Serializer):
+            raise serializers.ValidationError(_("Cannot expand a request for mosaicing with more than one configuration set"))
+        if request['configurations'][0]['target']['type'] != 'ICRS':
+            raise serializers.ValidationError(_("Mosaic expansion is only for ICRS Targets. Try using dither expansion for patterns with other target types"))
+
+        return request
+
+    def validate(self, data):
+        validated_data = super().validate(data)
+        # If point_overlap_percent is set, we will overwrite the point_spacing based on the requested
+        # instrument_type and its fov on horizontal axis. If line_overlap_percent is specified, we will
+        # do the same for the fov on the vertical axis - if it is not specified we will use the
+        # point_overlap_percent and overwrite the line_spacing value
+        if 'point_overlap_percent' in validated_data:
+            instrument_type = data['request']['configurations'][0]['instrument_type']
+            ccd_orientation = configdb.get_average_ccd_orientation(instrument_type)
+            pattern_orientation = validated_data.get('orientation', 0.0)
+            pattern_orientation = pattern_orientation % 360
+            # Decide to flip the point/line overlapped sense based on general orientation
+            if pattern_orientation < 45 or pattern_orientation > 315:
+                flip = False
+            elif pattern_orientation < 135:
+                flip = True
+            elif pattern_orientation < 225:
+                flip = False
+            elif pattern_orientation < 315:
+                flip = True
+            ccd_size = configdb.get_ccd_size(instrument_type)
+            pixel_scale = configdb.get_pixel_scale(instrument_type)
+            coso = cos(radians(ccd_orientation))
+            sino = sin(radians(ccd_orientation))
+            # Rotate the ccd dimensions by the ccd orientation - needed so our % overlap is in the correct frame
+            rotated_ccd_x = ccd_size['x'] * coso + ccd_size['y'] * sino
+            rotated_ccd_y = ccd_size['x'] * -sino + ccd_size['y'] * coso
+            if 'line_overlap_percent' not in validated_data:
+                validated_data['line_overlap_percent'] = validated_data['point_overlap_percent']
+            validated_data['point_spacing'] = abs(rotated_ccd_y) * pixel_scale * ((100.0 - validated_data['point_overlap_percent']) / 100.0)
+            validated_data['line_spacing'] = abs(rotated_ccd_x) * pixel_scale * ((100.0 - validated_data['line_overlap_percent']) / 100.0)
+            if flip:
+                # If the pattern orientation is closer to 90 or 270 (within 45 degrees), then flip the point/line spacing to better align with pattern orientation
+                temp = validated_data['line_spacing']
+                validated_data['line_spacing'] = validated_data['point_spacing']
+                validated_data['point_spacing'] = temp
+        elif 'point_spacing' not in validated_data:
```

Comment:
In the case that someone sets `point_spacing` instead of the `point_overlap_percent`, should we do something similar to line 916 and also set `line_spacing` to the provided `point_spacing` if it is unset?

Reply:
Thats actually already in the base PatternExpansionSerializer validate method - it will set the line_spacing to point_spacing if there is point_spacing and the pattern is grid. So we only need to worry about the overlap_percent fields in the MosaicSerializer.